rest/shared/src/main/scala/io/udash/rest/internal/UsesREST.scala

```diff
@@ -0,0 +1,126 @@
+package io.udash.rest.internal
+
+import com.avsystem.commons.rpc.{MetadataAnnotation, RPCMetadata}
+import io.udash.rest._
+import io.udash.rest.internal.RESTConnector.HTTPMethod
+
+import scala.collection.mutable
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.util.{Failure, Success}
+
+abstract class UsesREST[ServerRPCType](implicit val rpcMetadata: RPCMetadata[ServerRPCType],
+                                                     validRest: UdashRESTFramework#ValidREST[ServerRPCType],
+                                                     ec: ExecutionContext) {
+  val framework: UdashRESTFramework
+
+  import framework._
+
+  /**
+    * Proxy for remote RPC implementation. Use this to perform RPC calls.
+    */
+  lazy val remoteRpc = remoteRpcAsReal.asReal(new RawRemoteRPC(Nil))
+
+  /**
+    * This allows for generation of proxy which translates RPC calls into raw calls that
+    * can be sent through the network.
+    */
+  protected def remoteRpcAsReal: AsRealRPC[ServerRPCType]
+
+  protected val connector: RESTConnector
+
+  /** Transform `RawValue` into String used in HTTP request header. */
+  def rawToHeaderArgument(raw: RawValue): String
+  /** Transform `RawValue` into String used as HTTP request query argument. */
+  def rawToQueryArgument(raw: RawValue): String
+  /** Transform `RawValue` into String used as URL part. */
+  def rawToURLPart(raw: RawValue): String
```

Comment:
The default serialization uses them to remove quotes from the serialized `String`. In a request body it should look like `"string"`, but in a URL it should be `string`. In other serialization methods it might be even more useful.